### PR TITLE
feat(sql): implement jit for uuid type equality

### PR DESCRIPTION
**Pull request by eugenels**
This PR adds JIT support for the `UUID` type, utilising some assembler from Dave's PR: https://github.com/questdb/questdb/pull/2850
but with a slightly different approach.

The changes include:
- modifying the IR serialization format to support 16-byte payloads  
- direct UUID literals processing.
 
This enables the generation of AVX code for a commonly used filter expression: `"column = literal"`.
Only `eq` and `neq` operations currently implemented.
A separate PR will be needed to implement arithmetic operations for a `long128` type once the `long128` column type and literal are added.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 29 / 34 (85.29%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|29|34|85.29%|

**Comment by ideoma** [comment]
It looks good, but for a change like this, I'd expect 100% test coverage. Is there a good reason to not cover 5 lines?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 34 / 34 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|34|34|100.00%|

File: core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java [hunk 1]
```diff
@@ -580,13 +594,25 @@ private void serializeConstant(long offset, int position, final CharSequence tok
...
+                    throw SqlException.position(position).put("char constant in non-char expression: ").put(token);
+                }
                 // this is 'x' - char
                 putOperand(offset, IMM, I2_TYPE, token.charAt(1));
                 return;
+            } else if (len == 2 + Uuid.UUID_LENGTH) {
+                if (PredicateType.UUID != predicateContext.type) {
+                    throw SqlException.position(position).put("uuid constant in non-uuid expression: ").put(token);
+                }
+                final CharSequence uuid = token.subSequence(1, token.length() - 1);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java:608]: can we avoid creating sub-sequence, this is GC right there. Perhaps move this logic to `Uuid.parseLo()`. Give it offset to parse from ?

Comment (eugenels) [reply core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java]: Fixed in: https://github.com/questdb/questdb/pull/3032/commits/3b7517222d2b3c645fe0345b2b8597448d4d8c93

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 56 / 56 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/std/Uuid.java|22|22|100.00%|

**Review by ideoma — DISMISSED** [review]

**Review by bziobrowski — APPROVED** [review]
Looks good 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 56 / 56 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/std/Uuid.java|22|22|100.00%|

File: core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java]: 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 56 / 56 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/std/Uuid.java|22|22|100.00%|

**Pull request merged**